prov/rxm/src/rxm_conn.c

```diff
@@ -1275,14 +1273,17 @@ rxm_conn_handle_event(struct rxm_ep *rxm_ep, struct rxm_msg_eq_entry *entry)
 		}
 		reject_reason = cm_data->reject.reason;
 
-		if (reject_reason == RXM_CMAP_REJECT_GENUINE)
+		if (reject_reason == RXM_CMAP_REJECT_GENUINE) {
 			FI_WARN(&rxm_prov, FI_LOG_FABRIC, "connection reject: "
 			       "(reason: RXM_CMAP_REJECT_GENUINE)\n");
-		else if (reject_reason == RXM_CMAP_REJECT_SIMULT_CONN)
+			RXM_EQ_STRERROR(&rxm_prov, FI_LOG_WARN, FI_LOG_EP_CTRL,
+					rxm_ep->msg_eq, &entry->err_entry);
+		} else if (reject_reason == RXM_CMAP_REJECT_SIMULT_CONN) {
 			FI_DBG(&rxm_prov, FI_LOG_FABRIC, "connection reject: "
 			       "(reason: RXM_CMAP_REJECT_SIMULT_CONN)\n");
-		else
+		} else {
 			assert(0);
```

Comment:
If the reject_reason is coming from a peer, crashing the local app is a little harsh as well.

Reply:
I'll remove this and add a warning log instead.

Reply:
Okay, looking at the surrounding code, taking this path is a bug and we should have the assert. Once we verify that remote peer has sent a connection reject with correct cm_data version, the reason should be either of the two above. Otherwise it's a bug.

Reply:
Yes, but it's a bug coming from the peer.  Imagine having 2 versions of libfabric, where one adds a new reject code.  Maybe that should have updated the cm_data, but if it didn't, then the local node crashes, even though the 'bug' is from the peer.  That's my concern about trusting network data too much.